app/src/main/java/tech/relaycorp/gateway/domain/LocalConfig.kt

```diff
@@ -65,50 +69,71 @@ class LocalConfig
     }
 
     private suspend fun generateIdentityCertificate(privateKey: PrivateKey): Certificate {
-        val certificate = selfIssueCargoDeliveryAuth(privateKey, privateKey.toPublicKey())
+        val certificate = selfIssueCargoDeliveryAuth(
+            privateKey, privateKey.toPublicKey(), nowInUtc().plusYears(1)
+        )
         setIdentityCertificate(certificate)
         return certificate
     }
 
-    @Synchronized
     suspend fun bootstrap() {
-        try {
-            getIdentityKey()
-        } catch (_: RuntimeException) {
-            val keyPair = generateIdentityKeyPair()
-            generateIdentityCertificate(keyPair.private)
-        }
-
-        try {
-            getCargoDeliveryAuth()
-        } catch (_: RuntimeException) {
-            generateCargoDeliveryAuth()
+        mutex.withLock {
+            try {
+                getIdentityKey()
+            } catch (_: RuntimeException) {
+                val keyPair = generateIdentityKeyPair()
+                generateIdentityCertificate(keyPair.private)
+            }
+
+            getCargoDeliveryAuth() // Generates new CDA if non-existent
         }
     }
 
     suspend fun getCargoDeliveryAuth() =
-        fileStore.read(CDA_CERTIFICATE_FILE_NAME)
-            ?.let { Certificate.deserialize(it) }
-            ?: throw RuntimeException("No CDA issuer was found")
+        certificateStore.get()
+            .retrieveLatest(
+                getIdentityKey().privateAddress,
+                getIdentityCertificate().subjectPrivateAddress
+            )
+            ?.leafCertificate
+            .let { storedCertificate ->
+                if (storedCertificate?.isExpiringSoon() == false) {
+                    storedCertificate
+                } else {
+                    generateCargoDeliveryAuth()
+                }
+            }
+
+    suspend fun getAllValidCargoDeliveryAuth() =
+        certificateStore.get()
+            .retrieveAll(
+                getIdentityKey().privateAddress,
+                getIdentityCertificate().subjectPrivateAddress
+            )
+            .map { it.leafCertificate }
 
     private fun selfIssueCargoDeliveryAuth(
         privateKey: PrivateKey,
-        publicKey: PublicKey
+        publicKey: PublicKey,
+        expiryDate: ZonedDateTime
     ): Certificate {
         return issueGatewayCertificate(
             subjectPublicKey = publicKey,
             issuerPrivateKey = privateKey,
             validityStartDate = nowInUtc()
                 .minus(CalculateCRCMessageCreationDate.CLOCK_DRIFT_TOLERANCE.toJavaDuration()),
-            validityEndDate = nowInUtc().plusYears(1)
+            validityEndDate = expiryDate
```

Comment:
This doesn't need to match the `expiryDate` of the `getIdentityCertificate()` result. In fact, I think this should remain a constant value -- otherwise this would be one more place to change when we implement #556.

However, I think a year (as it was originally) is too long in hindsight. I think it should be 180 days for consistency with the desktop gateway -- although this, like all thresholds and limits in Awala, is mostly arbitrary until we have real-world usage. Thoughts?